libs/vkd3d/memory.c

```diff
@@ -1258,6 +1273,16 @@ static HRESULT vkd3d_memory_allocation_init(struct vkd3d_memory_allocation *allo
         }
     }
 
+    if (!(device->memory_info.upload_heap_memory_properties & VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT)
```

Comment:
This branch is impossible to read. Please split this up into somewhat readable booleans.